src/main/java/uk/nhs/digital/nhsconnect/nhais/mapper/PersonAddressMapper.java

```diff
@@ -0,0 +1,41 @@
+package uk.nhs.digital.nhsconnect.nhais.mapper;
+
+import org.hl7.fhir.r4.model.Address;
+import org.hl7.fhir.r4.model.Parameters;
+import org.hl7.fhir.r4.model.Patient;
+import org.hl7.fhir.r4.model.StringType;
+import uk.nhs.digital.nhsconnect.nhais.model.edifact.PersonAddress;
+
+import java.util.List;
+
+public class PersonAddressMapper implements FromFhirToEdifactMapper<PersonAddress> {
+    private final static Address.AddressUse ADDRESS_USE_HOME = Address.AddressUse.HOME;
+
+    public PersonAddress map(Parameters parameters) {
+        Address address = getAddress(parameters);
+
+        return PersonAddress.builder()
+            .addressLine1(getAddressLineOrNull(address.getLine(), 0))
+            .addressLine2(getAddressLineOrNull(address.getLine(), 1))
+            .addressLine3(getAddressLineOrNull(address.getLine(), 2))
+            .addressLine4(getAddressLineOrNull(address.getLine(), 3))
+            .addressLine5(getAddressLineOrNull(address.getLine(), 4))
+            .build();
+    }
+
+    private Address getAddress(Parameters parameters) {
+        Patient patient = getPatient(parameters);
+
+        return patient.getAddress().stream()
+            .filter(address -> address.getUse().equals(ADDRESS_USE_HOME))
+            .findFirst()
+            .orElseThrow(() -> new IllegalStateException("Address mapping problem"));
```

Comment:
Just leave `.orElseThrow()` in all of them

Reply:
If you're going to change it you need to throw a subclass NhaisBaseException so that API errors are produced correctly.

Don't use orElseThrow() we won't be getting helpful error messages that way.

Reply:
Did we even implemented error handling as of now?